Use `transform: translate()` instead of margin-* for the tooltips. Fix #483

wismill opened the pull request:
Fix #483 using CSS property `transform` instead of `margin-*` for the tooltips.

According to [caniuse.com](http://caniuse.com/#search=transform), this property has a very good support in the browsers.

seiyria commented:
this would need to pass our full CI suite before we could merge it. You'll also need to follow the contribution guidelines: https://github.com/seiyria/bootstrap-slider/blob/master/.github/CONTRIBUTING.md

You need to add a unit test to replace the one that was taken away, and I would like to see a jsfiddle of this.

wismill commented:
Well, it passes the full CI as I did not modify any of the lines reported by `codeclimate`.
I do follow your contribution guidelines. Please indicate what is not respected.
The unit test does not apply anymore, as the margin of the margins of the tooltips are never modified.

seiyria commented:
- It will need to pass code climate, regardless of whether you touched it or not
- You should add a jsfiddle so we can see it in action
- You should add a _new_ unit test in place of the old one.

wismill commented:
The issue detected by codeclimate is a duplication between javascript files in `src` and `dist`. How can I solve this?

seiyria commented:
I'm not sure why it would come up then. Maybe @rovolution knows why?

Either way though - the rest applies.

On Thu, Nov 2, 2017, 09:39 Wismill <notifications@github.com> wrote:

> The issue detected by codeclimate is a duplication between javascript
> files in src and dist. How can I solve this?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/seiyria/bootstrap-slider/pull/798#issuecomment-341442477>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAum2T-1O_0u_Bns55cbXBTp4LH7Tqxyks5sydQ8gaJpZM4QPs7->
> .
>

wismill commented:
JSFiddle:
- Original (see #483) http://jsfiddle.net/cwssL37d/
- With updated bootstrap-slider : http://jsfiddle.net/5waxge9b/1/

wismill commented:
Re-added the test, modified.

rovolution commented:
My one concern is that since [we state in our docs](https://github.com/seiyria/bootstrap-slider#supported-browsers) that we support IE9 and above, this will break that guarantee.

That being said, I would be ok with merging/releasing this, but would want to do so under a major version bump.

wismill commented:
It [is supported by IE 9](https://caniuse.com/#search=translate) using prefix `-ms-`. But this will add more code. Or all of this could be managed in CSS by using a class rather than a fixed CSS property. I'll check that.

wismill commented:
I have simplified the code and added support for IE 9. In fact, your code already uses CSS `transform` for `.slider-handle.triangle` and without compatibility.

rovolution commented:
> The issue detected by codeclimate is a duplication between javascript files in src and dist. How can I solve this?

Yea I agree this is a red-herring error. I'm going to try to find a way to ignore any changes to the dist directory for code climate checks

rovolution commented:
merged and released to v10.0.0 (published as a breaking change just incase there any inadvertent regressions). Thanks again for your contribution!
